### DISCUSSION TITLE
Delete a hung export request

### DISCUSSION THREAD

**Discussion by vgwidt** [outline/outline#3889] [General] [answered]
I have a hung export request after getting hit with a rate limit doing API requests.  The export that triggered it has been stuck in "creating" mode for nearly a day even after a reboot of the server.  I can create new export requests and they finish successfully:
![image](https://user-images.githubusercontent.com/49006240/181653060-87ba8c89-7be9-4fa1-a355-7255226b5c8e.png)

There is no delete option from the GUI, and attachments.delete doesn't appear to work for this.  Is there a way I can delete this from either the API or the GUI?

Also I realize this may be the result of a bug.  If so, I can collect more details and post to Issues if appropriate.  This is the rate limit exceeded error:
```
{"error":"Rate limit exceeded for this operation","stack":"TooManyRequestsError: Rate limit exceeded for this operation\n    at RateLimitExceededError (/opt/outline/build/server/errors.js:55:34)\n    at Function.checkRateLimit (/opt/outline/build/server/models/FileOperation.js:132:48)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Function.runHooks (/opt/outline/node_modules/sequelize/lib/hooks.js:94:7)\n    at async InstanceValidator._validateAndRunHooks (/opt/outline/node_modules/sequelize/lib/instance-validator.js:65:5)\n    at async InstanceValidator.validate (/opt/outline/node_modules/sequelize/lib/instance-validator.js:54:12)\n    at async Object.save (/opt/outline/node_modules/sequelize/lib/model.js:2368:7)\n    at async Object.update (/opt/outline/node_modules/sequelize/lib/model.js:2534:12)\n    at async ExportMarkdownZipTask.updateFileOperation (/opt/outline/build/server/queues/tasks/ExportMarkdownZipTask.js:107:5)\n    at async ExportMarkdownZipTask.perform (/opt/outline/build/server/queues/tasks/ExportMarkdownZipTask.js:87:7)\n    at async Queue.<anonymous> (/opt/outline/build/server/services/worker.js:132:7)","level":"error","message":"Error processing task in ExportMarkdownZipTask"}
```

**Comment by vgwidt** [comment, marked as answer]
I found "fileOperations.delete" and it worked!

I'm thinking it might be a good idea to allow for the deletion of a stuck export in the web interface, or perhaps an automatic deletion upon the rate limit being exceeded.

**Reply by thomaswennersten** [reply]
Hi, I´m wondering if you refer to a file or a command with fileOperations.delete?

**Reply by vgwidt** [reply]
It’s an API method (https://www.getoutline.com/developers). I’m not sure if there has since been a way to delete hung exports using the user interface though.

**Reply by thomaswennersten** [reply]
Where is this API call documented so that I can know how to use it? Can you provide an example of how you used it? I'm not seeing it at https://www.getoutline.com/developers#tag/FileOperations .".

**Reply by vgwidt** [reply]
Ah yeah it's not documented there.  I think when I had issues with hung exports, I'd get the list of operations using fileOperations.list to get the ID of the problem export.  Then you submit a POST request to the /api/fileOperations.delete with the body containing the ID of the operation you need to delete.  The API manual I posted should have details on how to use the API, in fact the curl example on top has the body you'd need to use.

Be careful with using delete though, and try to figure out what might be causing the hung export.  It sounds like the goal is to try to prevent hung exports completely rather than resort to having to delete them.

**Comment by tommoor** [comment]
> Also I realize this may be the result of a bug. 

Exactly, it should be impossible for a job to get "stuck" like this – so I'd rather figure that out than introduce functionality in the interface. Will try getting into a rate limit situation and see if that can reproduce it
